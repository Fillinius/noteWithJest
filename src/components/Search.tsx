import { ChangeEvent, useEffect, useMemo, useState } from 'react'
import { useDispatch } from 'react-redux'
import { addSearchToken } from 'src/store/searchSlice'

import { headerFieldOptions, validateHeaderMax } from 'src/utils/helpers'
// import './style.css'

export const Search = () => {
  const [search, setSearch] = useState('')
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(addSearchToken(search))
  }, [search])

  const hintValue = useMemo(
    () => (validateHeaderMax(search) ? '' : headerFieldOptions.message),
    [search]
  )

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    setSearch(event.target.value)
  }

  return (
    <div className="search-field-wrapper">
      <input
        type="text"
        value={search}
        onChange={handleChange}
        placeholder="search"
      />
      <span data-testid="search-hint-text" className="search-field-hint">
        {hintValue}
      </span>
    </div>
  )
}
