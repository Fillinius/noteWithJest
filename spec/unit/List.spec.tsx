import { render, screen } from '@testing-library/react'
import { List } from 'src/components/List'

const items: Task[] = [
  {
    id: '1',
    header: 'купить хлеб',
    done: true,
  },
  {
    id: '2',
    header: 'купить молоко',
    done: true,
  },
  {
    id: '3',
    header: 'выгулять собаку',
    done: true,
  },
  {
    id: '4',
    header: 'купить хлеб',
    done: true,
  },
  {
    id: '5',
    header: 'купить молоко',
    done: true,
  },
  {
    id: '6',
    header: 'выгулять собаку',
    done: true,
  },
  {
    id: '7',
    header: 'купить хлеб',
    done: true,
  },
  {
    id: '8',
    header: 'купить молоко',
    done: true,
  },
  {
    id: '9',
    header: 'выгулять собаку',
    done: true,
  },
  {
    id: '10',
    header: 'выгулять кошку',
    done: true,
  },
  {
    id: '11',
    header: 'выгулять cat',
    done: true,
  },
]

it('отображение списка задач', () => {
  const onDelete = jest.fn()
  const onToggle = jest.fn()

  const { rerender, asFragment } = render(
    <List items={items} onDelete={onDelete} onToggle={onToggle} />
  )
  const firstRender = asFragment()

  items.pop()

  rerender(<List items={items} onDelete={onDelete} onToggle={onToggle} />)
  const secondRender = asFragment()

  expect(firstRender).toMatchDiffSnapshot(secondRender)
})

it('Список содержит не больше 10 невыполненных задач', () => {
  const onDelete = jest.fn()
  const onToggle = jest.fn()
  render(<List items={items} onDelete={onDelete} onToggle={onToggle} />)
  const listEl = screen.queryAllByRole('button')
  expect(
    listEl.filter((item) => !item.hasAttribute('disabled')).length <= 10
  ).toBeTruthy()
})
