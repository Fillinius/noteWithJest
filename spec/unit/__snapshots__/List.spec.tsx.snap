// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображение списка задач 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -260,33 +260,7 @@
            alt="Удалить"
            src="[object Object]"
          />
        </button>
      </li>
-     <li
-       class="item-wrapper"
-     >
-       <input
-         checked=""
-         id="11"
-         type="checkbox"
-       />
-       <label
-         data-testid="label-field"
-         for="11"
-       >
-         <s>
-           выгулять cat
-         </s>
-       </label>
-       <button
-         class="button button-with-icon"
-         data-alt="Удалить"
-       >
-         <img
-           alt="Удалить"
-           src="[object Object]"
-         />
-       </button>
-     </li>
    </ul>
  </DocumentFragment>"
`;
